sui_types/src/messages.rs

```diff
@@ -314,14 +314,15 @@ where
 /// An transaction signed by a client. signature is applied on data.
 /// Any extension to Transaction should add fields to TransactionData, not Transaction.
 // TODO: this should maybe be called ClientSignedTransaction + SignedTransaction -> AuthoritySignedTransaction
-#[derive(Debug, Eq, Clone, Serialize, Deserialize)]
-pub struct Transaction {
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct TransactionEnvelop<S: AuthoritySignInfoTrait> {
```

Comment:
1) Missing e at end of `Envelop`
2) Maybe can call the trait `SignatureType` or `SignatureInfo` for short?  No biggie just slightly easier name

Reply:
Had to be verbose on the naming due to many other signature types in the same file.